### PR TITLE
Added clang-format and editorconfig

### PR DESCRIPTION
**Pull request by mrjvs**
references issue #8

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Language: Cpp
+
+AlignConsecutiveDeclarations: true
+ColumnLimit: 120
+IndentWidth: 4
+PointerAlignment: Right
+TabWidth: 4
+UseTab: ForIndentation
```
Comment (mrjvs) [inline .clang-format:8]: why no newline at end of file ;-;

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+[Makefile]
+indent_style = tab
+indent_size = 4
```
Comment (mrjvs) [inline .editorconfig:9]: I prefer 6 size

File: .clang-format (hunk 1 above)
Comment (pmerku) [reply .clang-format:8]: no need

File: .editorconfig (hunk 2 above)
Comment (pmerku) [reply .editorconfig:9]: you are a maniac

**Review by pmerku — APPROVED** [review]

**Pull request merged**